lib/elixir/lib/calendar/iso.ex

```diff
@@ -447,4 +463,139 @@ defmodule Calendar.ISO do
       {days + days_offset, {parts, ppd}}
     end
   end
+
+  # :calendar.date_to_gregorian_days(year, month, day) - 365
+  defp date_to_gregorian_days(year, month, day) do
+    last_day = days_in_month(year, month)
+    if day <= last_day do # TODO Why does :calendar have this check?
+      days_in_prev_years(year) + days_before_month(month) + leap_day_offset(year, month) + day - 1
+    end
```

Comment:
Yes. I have left it there for now because I tried to copy the implementation of `:calendar` 1:1, and this is [what is happening on line 118 of `:calendar`](https://github.com/erlang/otp/blob/master/lib/stdlib/src/calendar.erl#L118):

```
date_to_gregorian_days(Year, Month, Day) when is_integer(Day), Day > 0 ->
    Last = last_day_of_the_month(Year, Month),
    if
	Day =< Last ->
	    dy(Year) + dm(Month) + df(Year, Month) + Day - 1
    end.
```

We should definitely remove or alter it, or is the actual point in `:calendar` to return `nil` if it does not pass the check?

This internal function should only ever be called with `day` in the `0..366` range, so I don't think it should ever happen. Am I missing something?